### ISSUE TITLE
Tutorial on b96 dataset

### ISSUE THREAD

**Issue by sekeyWang** [novoalab/SeqTagger#4]
Hi,

I have successfully run the demo. However when I was trying to on the b96 dataset, I got confused since there's no tutorial on the dataset. From the dataset I can access the fast5 file, fastq files but I don't know what is the label(sequences or classified types) of these reads. Is it possible to give a simple tutorial about how to train and validate on the b96 dataset? Thanks!

Best,

Shaokai

**Comment by enovoa** [comment]
Dear @sekeyWang, 
The individual sequences for the 96 barcodes are listed [here](https://github.com/novoalab/SeqTagger/blob/main/models/b96_RNA002/barcodes.tsv)
The usage is the same as for the 4 barcodes, please replace the model in the seqtagger command line  for the 96 barcode model (-k models/b096_RNA002).
Best
Eva

**Comment by sekeyWang** [comment]
Hi Eva,

Thanks for your quick update! I was trying to run SeqTagger on EVA PRJEB78482 dataset. Is it possible to know the correct label of each read so that I can train and validate the model? Meanwhile, I didn't see test individual dataset for it. 

Best,

Shaokai

**Comment by GregorD96** [comment]
Hi Shaokai,

Thanks for using SeqTagger! 

If I understood correctly, you are interested in the correct labels for our b96 training data (RNA002) submitted to ENA under the accession PRJEB78482. This information can be retrieved from Supplemental Table S2 that can be downloaded [here](https://www.biorxiv.org/content/10.1101/2024.10.29.620808v1.supplementary-material). It provides you with the run name which corresponds to the name of the submitted .tar to ENA, as well as the barcode::sample relationship. The reference sequences for the IVT constructs can be found in this [repo](https://github.com/novoalab/SeqTagger/blob/main/ref/ref_ivt.fa).

I hope this is helpful.

Best,
Gregor

**Comment by sekeyWang** [comment]
Hi Gregor,

Thanks for your information! I have checked the labels in Table S2. If I understood correctly, for b96 training dataset, there are 4 types of barcode in each file and there are 96/4=16 files. Then in one file, the label for each read is still unknown since there are 4 barcode options. 

I have also looked into the fastq and bam file. It provides with a long sequence and seems not related to the DNA barcode. So I am still confused about it. I hope to see the output sequence of the trained bonito and its matching reference barcode for each read.

Best regards,

Shaokai

**Comment by GregorD96** [comment]
You are absolutely right, every file contains four barcodes but each barcode has been ligated to a different in vitro construct (also specified in Supplemental Table S2). 

By aligning your reads to the in vitro constructs you can infer which read contains which barcode.

Best,

Gregor

**Comment by sekeyWang** [comment]
Thanks Gregor! 
I think I am very close to understand. Could you explain a little bit more about how to align the reads to the barcode in one training dataset? For example given the fast5 file from RNA367508_scbc_1-4/20230712_1323_MN40194_FAW88460_0859045a, how can I get the corresponding correct barcode for each read?

Best regards,

Shaokai

**Comment by GregorD96** [comment]
to give you an example:

consider the following .tar file uploaded to ENA: **b04_RNA002_scbc_5_8_training**, from Supplemental Table S2 you can now retrieve the barcode to IVT relationship which in this case is:

SCBC-05 = IVT-09 (25s_5end_yrRNA_T7_BamHI)
SCBC-06 = IVT-10 (Lysine)
SCBC-07 = IVT-11 (pCDNA:3.1+:N-term:FLAG:TRMT1L)
SCBC-08 = IVT-12 (ALKBH5)

once you unzip the folder and move to the fast5 directory you can use a basecaller (guppy or dorado) to rebasecall the fast5 
 files. Next you'll need to align them to the reference sequences (linked above). 
Now you can extract from the bam file which read-ids map to each reference which will tell you the read::barcode relationship and go from there.

Please mind you that the run you specified RNA367508_scbc_1-4/20230712_1323_MN40194_FAW88460_0859045a which corresponds to **b04_RNA002_scbc_1_4_training** is not correct and should not be used. We are working on a fix.

Thanks and Best,

Gregor

**Comment by GregorD96** [comment]
Dear Shaokai,

Many apologies for the very late response.

We have since updated the runs on ENA, you'll find the correct run for  **b04_RNA002_scbc_1_4_training** under the following accession: [PRJEB84026](https://www.ebi.ac.uk/ena/browser/view/PRJEB84026?show=reads)

This should solve the problems you've encountered.

Once again, apologies for the late reply.

Best,

Gregor
